packages/contrast-colors/index.js

```diff
@@ -13,7 +13,480 @@ governing permissions and limitations under the License.
+          swatchNames = Object.keys(color.ratios);
+          ratioValues = Object.values(color.ratios);
+        }
+
+        // modify target ratio based on contrast multiplier
+        let newRatioValues = ratioValues.map(ratio => multiplyRatios(ratio, this._contrast) );
+        if(this._contrast !==1) ratioValues = newRatioValues;
+
+        // Return color matching target ratio, or closest number
+        for (let i=0; i < ratioLength; i++){
+          // Find the index of each target ratio in the array of all possible contrasts
+          let r = getMatchingRatioIndex(contrasts, ratioValues[i]);
+          let match = color.colorScale[r];
+
+          // Use the index from matching contrasts (r) to index the corresponding
+          // color value from the color scale array.
+          // use convertColorValue function to convert each color to the specified 
+          // output format and push to the new array 'contrastColors'
+          contrastColors.push(convertColorValue(match, this._output));
+        }
+
+        for (let i=0; i < contrastColors.length; i++) {
+          let n;
+          if(!swatchNames) {
+            let rVal = ratioName(color.ratios)[i];
+            n = color.name.concat(rVal);
+          }
+          else {
+            n = swatchNames[i];
+          }
+  
+          let obj = {
+            name: n,
+            contrast: ratioValues[i],
+            value: contrastColors[i]
+          };
+          newArr.push(obj);
+          // Push the same value to the returnColorValues array
+          returnColorValues.push(contrastColors[i]);
+
+        }
+        returnColors.push(colorObj);
+      }
+    });
+    this._contrastColorValues = returnColorValues;
+    this._contrastColors = returnColors;
+    return this._contrastColors;
+  }
+
+  _findContrastColorValues() {
+    return this._contrastColorValues;
+  }
+}
+
+class Color {
+  constructor({name, colorKeys, colorspace = 'RGB', ratios, smooth = false, output = 'HEX'}) { 
+    this._name = name;
+    this._colorKeys = colorKeys;
+    this._colorspace = colorspace;
+    this._ratios = ratios;
+    this._smooth = smooth;
+    this._output = output;
+    if (!this._name) {
+      throw new Error('Color missing name');
+    } 
+    if (!this._colorKeys) {
+      throw new Error(`Color Keys are undefined`);
+    }
+    if (!colorSpaces[this._colorspace]) {
+      throw new Error(`Colorspace “${colorspace}” not supported`);
+    }
+    if (!colorSpaces[this._output]) {
+      throw new Error(`Output “${colorspace}” not supported`);
+    }
+    // validate color keys
+    for (let i=0; i<this._colorKeys.length; i++) {
+      if (this._colorKeys[i].length < 6) {
+        throw new Error('Color Key must be greater than 6 and include hash # if hex.');
+      }
+      else if (this._colorKeys[i].length == 6 && this._colorKeys[i].charAt(0) != 0) {
+        throw new Error('Color Key missing hash #');
+      }
+    }
```

Comment:
It seems odd to assume something is hex based on length, I could pass `#333` which is valid hex and your check would reject it, or `333` which is invalid hex, and your check here would reject it, but for the wrong reason.

Also, is `orange` not valid input? It's a color name that's 6 chars long but doesn't start with a `#`.